### PR TITLE
Add tests for `ReadOnlyRun`

### PR DESCRIPTION
**Pull request by kgodlewski**
This is an initial batch of tests that don't rely on prepopulated data, but instead log them during test execution.

In order not to interfere with existing tests that do rely on runs in a project having a specific shape, it is possible to point selected test modules to another existing project. This is already configured in GitHub Actions variables and secrets.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
+
+## Environment variables
+
+* `NEPTUNE_PROJECT` - project name to use. It is set by the test runner script in
+  GitHub setup.
+* `NEPTUNE_API_TOKEN` - API token to use
+
+Test that populate data during execution could interfere with those that rely on a
+specific data shape. To avoid this, set `NEPTUNE_E2E_PROJECT` to a different project
+that already exists.
```
Comment (michalsosn) [inline tests/README.md:28]: so, NEPTUNE_PROJECT - tests on prepopulated with data, NEPTUNE_E2E_PROJECT - tests on a specific data shape 
or is the opposite? I don't think it's unambiguous in this description, nor is it clear from the variable names

File: tests/e2e/conftest.py [hunk 2]
```diff
@@ -50,3 +68,68 @@ def id_to_name():
...
+    def _wrap(self, fn: Callable):
+        def wrapped(*args, **kwargs):
+            result = fn(*args, **kwargs)
+            self.run.wait_for_processing()
+            return result
+
+        wrapped._is_wrapped = True
+        setattr(self.run, fn.__name__, wrapped)
+
+        return wrapped
```
Comment (michalsosn) [inline tests/e2e/conftest.py:99]:
```
class SyncRun(Run):
    def log(self, *args, **kwargs):
        result = super().log(*args, **kwargs)
        self.wait_for_processing()
        return result
```
hmm, would this code achieve the same thing and be somewhat simpler?

Comment (kgodlewski) [reply tests/e2e/conftest.py:99]: You're absolutely right! It requires some modification to the tests, but I'll go with it. I started with the wrapper approach for other methods as well, and ended up only needing log*(), so your solution does the thing and is indeed simpler.

File: tests/README.md (hunk 1 above)
Comment (kgodlewski) [reply tests/README.md]: I changed both the descriptions and variable names for clarity

**Review by michalsosn — APPROVED** [review]

**Review by PatrykGala — CHANGES_REQUESTED** [review]
rebase to main

**Review by PatrykGala — APPROVED** [review]

**Review by PatrykGala — APPROVED** [review]

**Pull request merged**